File: perl/domain-reputation-api.pl
#!/usr/bin/perl

use JSON qw( decode_json );       # From CPAN
use LWP::UserAgent;               # From CPAN
use URI::Escape qw( uri_escape ); # From CPAN

use strict;
use warnings;

my $api_key = 'your domain reputation scoring api key';
my $base_url = 'https://api.threatintelligenceplatform.com/v1/reputation';
my $domain_name = 'threatintelligenceplatform.com';

my $url = $base_url
        . '?apiKey=' . uri_escape($api_key)
        . '&domainName=' . uri_escape($domain_name);

print 'Querying the API: ', $url, '...', "\n\n";

my $ua = LWP::UserAgent->new;
my $response = $ua->get($url);

die $response->status_line unless $response->is_success;

my $decoded_json = decode_json($response->content);
my $score = $decoded_json->{'reputationScore'};

print 'Domain name: ', $domain_name, "\n";
print 'Score: ', ($score) ? $score : 'Something went wrong.', "\n";

File: python/domain-reputation-api.py
try:
    from urllib.request import urlopen, pathname2url
except ImportError:
    from urllib import pathname2url
    from urllib2 import urlopen

import json


def print_response(txt):
    response_json = json.loads(txt)
    print('Reputation score: ' + str(response_json['reputationScore']))


api_key = 'Your Domain Reputation Scoring API key'
domain = 'threatintelligenceplatform.com'
mode = 'fast'

url = 'https://api.threatintelligenceplatform.com/v1/reputation'\
    + '?apiKey=' + pathname2url(api_key)\
    + '&domainName=' + pathname2url(domain)\
    + '&mode' + pathname2url(mode)

try:
    response = urlopen(url).read().decode('utf8')
    print_response(urlopen(url).read().decode('utf8'))
except Exception as e:
    print(e)
